### DISCUSSION TITLE
UNet deepest layer number of features

### DISCUSSION THREAD

**Discussion by kissievendor** [Project-MONAI/MONAI#1760] [Q&A] [answered]
Hi,

I was again looking at the UNet architecture, and I noticed that there is an extra convolution (stride 1x1x1) in the deepest layer. 
So, if I define my network like below, the 256 features is achieved through convolution of 128 instead of strided convolution. 

Could you explain the reasoning behind this?

Thank you. 

model = UNet(
    dimensions=3,
    in_channels=1,
    out_channels=7,
    kernel_size=3,
    channels=(32, 64, 128,256),
    strides=(2,2,2),
    norm=Norm.BATCH,
    dropout=0.2,
).to(device)

edit: 
I have added the output of this model (print(model)). It seems that in the deepest layer, the connection is both a convolution and a skip connection. 

![model_base_print](https://user-images.githubusercontent.com/54887561/110914092-35333680-8316-11eb-8adb-1fd5a78f3ce5.PNG)

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

Could you please help share some info here?

Thanks.

**Comment by ericspod** [comment, marked as answer]
The final value in the `channels` argument defines the size of the bottom layer of the UNet structure, which doesn't involve any down/upsampling thus the stride of 1. UNet typically has a bottom layer of convolutions representing information dense in the channel dimension with downsampled spatial dimensions, this is the latent space the upsampling branch of the network starts with to decode the output.

**Comment by kissievendor** [comment]
Okay, so its goes from 128 with a normal convolution to 256. Where does the SkipConnection come from (384)? I was under the impression that after the last down-sampling, so in the bottom layer, it would go to up-sampling directly. So without the convolution with stride 1.

**Comment by ericspod** [comment]
That follows the bottom layer and is the first layer of the upsampling branch. The UNet is structured as embedded layers where each includes the downsample path, a skip connection, the next layer down, and the upsample branch concatenating data from the skip connection and the output of the next layer down.

**Comment by kissievendor** [comment]
Thank you for answering my questions. I think I understand it now.
